Settings.c

```diff
@@ -603,11 +603,13 @@ static void writeMeterModes(const Settings* this, FILE* fd, char separator, unsi
 int Settings_write(const Settings* this, bool onCrash) {
    FILE* fd;
    char separator;
+   char *filename;
    if (onCrash) {
       fd = stderr;
       separator = ';';
    } else {
-      fd = fopen(this->filename, "w");
+      xAsprintf(&filename, "%s.tmp", this->filename);
+      fd = fopen(filename, "w");
```

Comment:
May I suggest using [mkstemp](https://man7.org/linux/man-pages/man3/mkstemp.3.html) for the purpose or is it an overkill?
```c
      xAsprintf(&filename, "%s.XXXXXX", this->filename);
      fd = mkstemp(filename);
```

Reply:
Not exactly overkill, but when multiple instances of htop are running and quit quickly after each other you would still run into an race of which configuration actually prevails. But avoiding the possible corruption of multiple instances writing the same temp file might be worth another look at this part (you then at least know the surviving configuration has not been corrupted while it was generated) …